src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/MetricItemExtensions.cs

```diff
@@ -267,37 +268,12 @@ internal static OtlpMetrics.Metric ToOtlpMetric(this Metric metric)
                             }
                         }
 
-                        var exemplars = metricPoint.GetExemplars();
-                        foreach (var examplar in exemplars)
+                        if (metricPoint.TryGetExemplars(out var exemplars))
                         {
-                            if (examplar.Timestamp != default)
+                            foreach (ref readonly var exemplar in exemplars)
                             {
-                                byte[] traceIdBytes = new byte[16];
-                                examplar.TraceId?.CopyTo(traceIdBytes);
-
-                                byte[] spanIdBytes = new byte[8];
-                                examplar.SpanId?.CopyTo(spanIdBytes);
-
-                                var otlpExemplar = new OtlpMetrics.Exemplar
-                                {
-                                    TimeUnixNano = (ulong)examplar.Timestamp.ToUnixTimeNanoseconds(),
-                                    TraceId = UnsafeByteOperations.UnsafeWrap(traceIdBytes),
-                                    SpanId = UnsafeByteOperations.UnsafeWrap(spanIdBytes),
-                                    AsDouble = examplar.DoubleValue,
-                                };
-
-                                if (examplar.FilteredTags != null)
-                                {
-                                    foreach (var tag in examplar.FilteredTags)
-                                    {
-                                        if (OtlpKeyValueTransformer.Instance.TryTransformTag(tag, out var result))
-                                        {
-                                            otlpExemplar.FilteredAttributes.Add(result);
-                                        }
-                                    }
-                                }
-
-                                dataPoint.Exemplars.Add(otlpExemplar);
+                                dataPoint.Exemplars.Add(
+                                    ToOtlpExemplar(exemplar.DoubleValue, in exemplar));
```

Comment:
Out of curiosity, why do we only process `DoubleValue`? Shouldn’t we check whether `LongValue` or `DoubleValue` has a value?

Reply:
Never mind, got answer myself. It comes from valuestorage and we type cast accordingly.

Reply:
It's because Histogram updates are always tracked as `double` values: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/MetricPoint.cs#L409